### DISCUSSION TITLE
RequestValidationError, ErrorWrapper and ValidationError under Pydantic V2

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [fastapi/fastapi#10076] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
see description
```

### Description

I posted some of this on Pydantic discussions but the question belongs here because a FastApi centric solution is needed. I have a FastApi application (w/ Pydantic V1) and want to migrate it to FastApi v0.101.0+ (with Pydantic V2). My (V1)  application error handling successfully uses this construct:
```
from fastapi.exceptions import HTTPException, RequestValidationError
from pydantic import ErrorWrapper

try:
    this = something('odd')
except:
    raise RequestValidationError(errors=[ErrorWrapper(ValueError('Invalid'), 'odd')])
```
With FastApi v0.101 and Pydantic V2, and replacing ErrorWrapper with ValidationError, the above code causes a TypeError:
```
from fastapi.exceptions import HTTPException, RequestValidationError
from pydantic import ValidationError

try:
    this = something('odd')
except:
    raise RequestValidationError(errors=[ValidationError(ValueError('Invalid'), 'odd')])

TypeError: No constructor defined
```

The Pydantic suggestion is to use ValidationError.from_exception_data like this (I think), which also causes a TypeError:
```
from fastapi.exceptions import HTTPException, RequestValidationError
from pydantic import ValidationError

try:
    this = something('odd')
except:
    raise RequestValidationError(errors=[ValidationError.from_exception_data(ValueError('Invalid'), 'odd')])

TypeError: argument 'line_errors': 'str' object cannot be converted to 'PyList'
```
I use mypy for type checking and have also looked at https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core._pydantic_core.ValidationError.from_exception_data but am none the wiser :(

What am I not understanding and what is the way forward? Thanks

### Operating System

Windows

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11.4

**Comment by acdha** [comment, marked as answer]
https://github.com/tiangolo/fastapi/discussions/8033#discussioncomment-6532805 has a working example
